modules/auxiliary/cloud/aws/enum_ssm.rb

```diff
@@ -0,0 +1,153 @@
+                          target-level credentials). The module also provides an EC2 ID
+                          filter and a limiting throttle to prevent session stampedes or
+                          expensive messes.
+                         ),
+        'Author'      => [
+          'Aaron Soto <aaron.soto@rapid7.com>', # EC2 enum module
+          'RageLtMan <rageltman[at]sempervictus>' # SSM stuff
+        ],
+        'License'     => MSF_LICENSE
+      )
+    )
+
+    register_options(
+      [
+        OptBool.new('GET_SSM_SESSION', [true, 'Automatically get SSM sessions once found', false]),
+        OptInt.new('LIMIT', [false, 'Only return the specified number of results from each region']),
+        OptString.new('FILTER_EC2_ID', [false, 'Look for specific EC2 instance ID']),
+        OptString.new('REGION', [false, 'AWS Region (eg. "us-west-2")']),
+        OptString.new('ACCESS_KEY_ID', [true, 'AWS Access Key ID (eg. "AKIAXXXXXXXXXXXXXXXX")', '']),
+        OptString.new('SECRET_ACCESS_KEY', [true, 'AWS Secret Access Key (eg. "CA1+XXXXXXXXXXXXXXXXXXXXXX6aYDHHCBuLuV79")', ''])
+      ]
+    )
+  end
+
+  def handle_aws_errors(e)
+    if e.class.module_parents.include?(Aws)
+      fail_with(Failure::UnexpectedReply, e.message)
+    else
+      raise e
+    end
+  end
+
+  def enumerate_regions
+    regions = []
+
+    ec2 = Aws::EC2::Resource.new(
+      region: 'us-east-1',
+      access_key_id: datastore['ACCESS_KEY_ID'],
+      secret_access_key: datastore['SECRET_ACCESS_KEY']
+    )
+
+    ec2_regions = ec2.client.describe_regions.data.regions
+    ec2_regions.each do |r|
+      regions.append(r.region_name)
+    end
+
+    regions
+  end
+
+  def run
+    regions = datastore['REGION'] ? [datastore['REGION']] : regions = enumerate_regions()
+    credentials = ::Aws::Credentials.new(datastore['ACCESS_KEY_ID'], datastore['SECRET_ACCESS_KEY'])
+    regions.each do |region|
+      begin
+        vprint_status "Checking #{region}..."
+        client = ::Aws::SSM::Client.new(
+          region: region,
+          credentials: credentials,
+        )
+        inv_params = { filters: [
+          {
+            key: "AWS:InstanceInformation.InstanceStatus",
+            values: ["Terminated"],
+            type: "NotEqual",
+          },
+          {
+            key: "AWS:InstanceInformation.ResourceType",
+            values: ['EC2Instance'],
+            type: "Equal",
+          }
+        ]}
+
+        inv_params[:filters] << {
+          key: "AWS:InstanceInformation.InstanceId",
+          values: [datastore['FILTER_EC2_ID']],
+          type: "Equal",
+        } if datastore['FILTER_EC2_ID']
+
+        ssm_ec2 = client.get_inventory(inv_params).entities.map {|e| e.data["AWS:InstanceInformation"].content}.flatten
+        ssm_ec2 = ssm_ec2[0...datastore['LIMIT']] if datastore['LIMIT']
+        ssm_ec2.each do |ssm_host| 
+          vprint_good JSON.pretty_generate(ssm_host)
```

Comment:
It'd be nice if this printed something a bit more readable and didn't look like raw output.

Reply:
This is one of the things i left open to "those who know current standards" - its pure PoC-ware right now. Probably want it to report hosts and table-print the outputs. Wanna give me spec for what should be done or PR your take on it?